qiskit/circuit/instruction.py

```diff
@@ -405,17 +405,30 @@ def inverse(self):
 
     def c_if(self, classical, val):
         """Set a classical equality condition on this instruction between the register or cbit
-        ``classical`` and value ``val``.
+        or list of cbits, ``classical`` and value ``val``.
 
         .. note::
 
             This is a setter method, not an additive one.  Calling this multiple times will silently
             override any previously set condition; it does not stack.
         """
-        if not isinstance(classical, (ClassicalRegister, Clbit)):
-            raise CircuitError("c_if must be used with a classical register or classical bit")
+        if not isinstance(classical, (ClassicalRegister, Clbit)) and not all(
+            isinstance(cbit, Clbit) for cbit in classical
+        ):
+            raise CircuitError(
+                "c_if must be used with a classical register, a bit or a list of classical bits"
+            )
         if val < 0:
             raise CircuitError("condition value should be non-negative")
+
+        if (isinstance(classical, ClassicalRegister) and val >= 2**classical.size) or (
+            isinstance(classical, list) and val >= 2 ** len(classical)
+        ):
+            raise CircuitError("condition value should be less than 2 ^ number of bits")
+
+        if isinstance(classical, Clbit) and int(val) > 1:
+            raise CircuitError("condition value should be 0/1 or True/False")
```

Comment:
I tried to trigger this exception (in order to create a test for it). But it seems like it is covered by line 421 
> raise CircuitError("condition value should be non-negative")

Reply:
Triggered! testing in 9cafc8571

Still not fully sure how 421 does not get this one.